tests/test_messages.py

```diff
@@ -27,101 +33,129 @@ class WhiteCat(BaseModel):
 class DeepBody(BaseModel):
     deep_body: str = Field(..., min_length=20)
     url: AnyHttpUrl
+    ipv4: IPv4Address
+
+    @field_validator('url')
+    @classmethod
+    def validate_url(cls, v: AnyHttpUrl):
+        return v.__str__()
+
+    @field_validator('ipv4')
+    @classmethod
+    def validate_ipv4(cls, v: IPv4Address):
+        return v.__str__()
 
 
 class InnerNestedExample(BaseModel):
-    inner_body: List[DeepBody] = Field(..., max_items=2)
+    inner_body: List[DeepBody] = Field(..., max_length=2)
     cat: Cat
+    ipv6: IPv6Address
+
+    @field_validator('ipv6')
+    @classmethod
+    def validate_ipv6(cls, v: IPv6Address):
+        return v.__str__()
+
+
+CustomType = Annotated[str | int, Field(..., min_length=2, max_length=5)]
 
 
 class NestedExample(BaseModel):
     body: str = Field(..., max_length=2)
     inner: InnerNestedExample
+    custom_input: CustomType
+
+
+class MyEnum(str, Enum):
+    option = 'option'
+    choice = 'choice'
 
```

Comment:
**Suggestion:** Avoid using magic strings. Consider defining 'option' and 'choice' as constants.
```suggestion
OPTION = 'option'
CHOICE = 'choice'

class MyEnum(str, Enum):
    option = OPTION
    choice = CHOICE
```